### PR TITLE
Delete token and redirect to login if error status is 401

### PR DESCRIPTION
**Pull request by maryna-velychko**

### CODE REVIEW

File: src/app/Header.js [hunk 1]
```diff
@@ -59,17 +60,29 @@ function NavLink({link}) {
 
 function LogOutLink() {
   const theme = useTheme()
+  const router = useRouter();
   const { enqueueSnackbar } = useSnackbar();
   const { setAuthToken } = useContext(AuthTokenContext);
+
```
Comment (shuryhin-oleksandr) [inline src/app/Header.js:66]: It does not cover a case, when I am trying to access some page with an invalid token, it covers only logout link. It shourd redirect all 401 requests to login.

File: src/app/api.mjs [hunk 2]
```diff
@@ -32,12 +32,28 @@ const client = axios.create({
...
-  return config;
-});
+  
+client.interceptors.request.use(
+  (config) => {
+    const authToken = localStorage.getItem(AUTH_TOKEN_KEY);
+    if (authToken) config.headers.Authorization = `Token ${authToken}`;
+    return config;
+  },
+  (error) => {
```
Comment (shuryhin-oleksandr) [inline src/app/api.mjs:42]: Why do we need this?

File: src/app/api.mjs [hunk 3]
```diff
@@ -32,12 +32,28 @@ const client = axios.create({
...
+    return Promise.reject(error);
+  }
+);
+
+client.interceptors.response.use(
+  (response) => response,
```
Comment (shuryhin-oleksandr) [inline src/app/api.mjs:48]: Can we omit this line?

File: src/app/api.mjs [hunk 4]
```diff
@@ -32,12 +32,28 @@ const client = axios.create({
...
+  (error) => {
+    if (error.response?.status === 401) {
+      localStorage.removeItem(AUTH_TOKEN_KEY);
+      window.location.href = '/login';
```
Comment (shuryhin-oleksandr) [inline src/app/api.mjs:52]: Why do not use Next router?

File: src/app/api.mjs (hunk 3 above)
Comment (maryna-velychko) [reply src/app/api.mjs:48]: no, it is not working without it

File: src/app/api.mjs (hunk 2 above)
Comment (maryna-velychko) [reply src/app/api.mjs]: sorry, I took it from the documentation

**Pull request merged**